spark/src/test/scala/geotrellis/spark/regrid/Regrid.scala

```diff
@@ -0,0 +1,69 @@
+package geotrellis.spark.regrid
+
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.testkit._
+import geotrellis.spark._
+import geotrellis.spark.testkit._
+import geotrellis.spark.tiling._
+import geotrellis.vector._
+
+import org.scalatest._
+
+class RegridSpec extends FunSpec with TestEnvironment with RasterMatchers {
+
+  val simpleLayer = {
+    val tiles = 
+      for ( x <- 0 to 3 ;
+            y <- 0 to 2
+      ) yield {
+        val tile = IntArrayTile.ofDim(32, 32)
+        (SpatialKey(x, y), tile.map{ (tx, ty, _) => math.max(tx + 32 * x, ty + 32 * y) })
+      }
+    val rdd = sc.parallelize(tiles)
+    val ex = Extent(0,0,12.8,9.6)
+    val ld = LayoutDefinition(GridExtent(ex, 0.1, 0.1), 32, 32)
+    val md = TileLayerMetadata[SpatialKey](IntConstantNoDataCellType,
+                                           ld,
+                                           ex,
+                                           LatLng,
+                                           KeyBounds[SpatialKey](SpatialKey(0,0), SpatialKey(3,2)))
+    ContextRDD(rdd, md)
+  }
+
+  describe("Regridding") {
+    it("should allow chipping into smaller tiles") {
+      val newLayer = simpleLayer.regrid(16)
+
+      // import geotrellis.raster.render._
```

Comment:
Mb tests cleanup / more tests are required here?

Reply:
Willing to take suggestions about tests that will cover cases that I don't.  These tests check what happens when we evenly break the source into smaller units, coalesce tiles into larger units, and when we extend in one axis and contract in the other (also covering the case when the target domain's extent is larger than the source domain).  I can't think of what I might be missing.